query-node/schema.graphql

```diff
@@ -386,3 +386,22 @@ type FeaturedVideo @entity {
   "Reference to a video"
   video: Video!
 }
+
+enum StorageProviderType {
+  GATEWAY
+  STORAGE
+}
+
+type StorageProvider @entity {
+  "Runtime entity identifier (EntityId)"
+  id: ID!
```

Comment:
This will not work.. You need separate `@entity` for each group. because worker Id  is NOT unique across the various groups. so you can have a storage provider with id 10, as well as gateway worker with the same id 10.

Reply:
All workers are still saved into one `@entity`, but they save runtime id into `workerId`. `workerId` in combination with `workerType` can be used to distinguish workers from different working groups.

Reply:
Example GraphQL query: 
```
query {
   workers(where: {workerId_eq: "1", type_eq: STORAGE }) {id, workerId type}
}
```